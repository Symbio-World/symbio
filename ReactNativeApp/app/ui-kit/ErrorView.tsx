import React from 'react'
import { View, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'

type Props = {
  error?: unknown
}
export const ErrorView: React.FC<Props> = () => {
  return (
    <View style={[t.flex1, t.justifyCenter]}>
      <Text>Something went wrong...</Text>
    </View>
  )
}
