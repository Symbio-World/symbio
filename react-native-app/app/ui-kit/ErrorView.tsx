import * as React from 'react'
import { View, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { RFPercentage } from 'react-native-responsive-fontsize'

import { Icon, Icons } from './Icon'
import { CloseButtonView } from './CloseButtonView'
import { saveError } from '../error-handler/saveError'

type Props = {
  error?: Error
  onClose?: () => void
}

export const ErrorView: React.FC<Props> = ({ onClose = () => {}, error }) => {
  React.useEffect(() => {
    if (error) {
      saveError(error.message)
    }
  })
  return (
    <View style={[t.flex1]}>
      <Icon icon={Icons.FATAL_ERROR} />
      <Text
        style={[
          t.textCenter,
          t.textBlack,
          t.mB10,
          { fontSize: RFPercentage(7) },
        ]}
      >
        OOOOPS!
      </Text>
      <Text
        style={[
          t.textCenter,
          t.textBlack,
          t.mX12,
          { fontSize: RFPercentage(4) },
        ]}
      >
        Something went wrong... You can tell us what happened or try again later
      </Text>
      <CloseButtonView onClose={onClose} />
    </View>
  )
}
