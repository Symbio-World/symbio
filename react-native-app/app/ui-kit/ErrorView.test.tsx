import React from 'react'
import { render } from 'react-native-testing-library'
import { ErrorView } from './ErrorView'

describe('ErrorView', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<ErrorView />)
    expect(toJSON()).toMatchSnapshot()
  })
})
