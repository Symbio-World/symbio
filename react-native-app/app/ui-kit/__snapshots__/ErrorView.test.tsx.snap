// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <SvgMock />
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#000",
        },
        Object {
          "marginBottom": 40,
        },
        Object {
          "fontSize": 93,
        },
      ]
    }
  >
    OOOOPS!
  </Text>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#000",
        },
        Object {
          "marginHorizontal": 48,
        },
        Object {
          "fontSize": 53,
        },
      ]
    }
  >
    Something went wrong... You can tell us what happened or try again later
  </Text>
  <View
    style={
      Array [
        Object {
          "height": 160,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "bottom": 0,
        },
        Object {
          "width": "100%",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "paddingBottom": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderWidth": 2,
            },
            Object {
              "borderColor": "#B35340",
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "justifyContent": "center",
            },
            Object {
              "width": 64,
            },
            Object {
              "height": 64,
            },
            Object {
              "borderRadius": 9999,
            },
            Object {
              "padding": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 20,
              },
              Object {
                "height": 20,
              },
            ]
          }
        >
          <SvgMock
            height="100%"
            width="100%"
          />
        </View>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "fontSize": 20,
              },
              Object {
                "color": "#B35340",
              },
            ]
          }
        >
          close
        </Text>
      </View>
    </View>
  </View>
</View>
`;
